### PR TITLE
fix: Parse plugin logs coming out during `meltano invoke`

### PR DESCRIPTION
**Pull request by edgarrmondragon**
## Related Issues

* Closes #XXXX

## Summary by Sourcery

Improve `meltano invoke` plugin log handling by parsing structured stderr output and aligning plugin directory resolution with the new project directory API.

Enhancements:
- Add a log output handler to parse plugin stderr using configured log parsers during `meltano invoke`, falling back gracefully to plain logging when parsing fails.
- Capture plugin stderr asynchronously in `meltano invoke` and route it through structured logging with proper log levels and extra fields preserved.
- Update plugin configuration, run, and virtualenv directory resolution to use the newer `project.dirs` helpers instead of deprecated path accessors.

Tests:
- Add unit tests for the log output handler covering structured Singer SDK logs, plain text logs, empty lines, multiple severity levels, and preservation of extra fields.

### CODE REVIEW

File: src/meltano/cli/invoke.py [hunk 1]
```diff
@@ -191,9 +242,34 @@ async def _invoke(  # noqa: ANN202
...
+                    stdio="stderr",
+                    cmd_type="command",
+                )
+
+                # Create output handler for stderr
+                stderr_out = _LogOutputHandler(stderr_logger, log_parser=log_parser)
+
+                # Capture stderr output asynchronously
+                stderr_future = asyncio.create_task(
+                    capture_subprocess_output(handle.stderr, stderr_out),
```
[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply src/meltano/cli/invoke.py:266]: Ok, got a code example?

[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply src/meltano/cli/invoke.py:266]: Ok, got a code example?

[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply src/meltano/cli/invoke.py:266]: Ok, got a code example?

File: src/meltano/cli/invoke.py [hunk 2]
```diff
@@ -158,6 +160,55 @@ async def invoke(
...
+            extra = {**parsed_record.extra}
+            if parsed_record.logger_name:
+                extra["plugin_logger"] = parsed_record.logger_name
+
+            self.logger.log(
+                parsed_record.level,
+                parsed_record.message,
+                plugin_exception=parsed_record.exception,
+                **extra,
+            )
```
[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply src/meltano/cli/invoke.py:206]: We're good

File: tests/meltano/cli/test_invoke.py [hunk 3]
```diff
@@ -268,3 +271,107 @@ def test_invoke_only_install(self, cli_runner, project: Project, utility) -> Non
...
+    def test_writeline_preserves_extra_fields(self, caplog):
+        """Test that extra fields from parsed logs are preserved."""
+        logger = structlog.stdlib.get_logger("test")
+        handler = _LogOutputHandler(logger, log_parser="singer-sdk")
+
+        log_with_extras = (
+            '{"level":"info","pid":123,"logger_name":"tap-test","ts":1234567890.0,'
+            '"thread_name":"MainThread","app_name":"singer-sdk","stream_name":"users",'
+            '"message":"Processing stream","extra":{"record_count":100}}'
+        )
```
[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply tests/meltano/cli/test_invoke.py]: PTAL: 722798b02

[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply tests/meltano/cli/test_invoke.py]: So?

File: tests/meltano/cli/test_invoke.py [hunk 4]
```diff
@@ -268,3 +271,149 @@ def test_invoke_only_install(self, cli_runner, project: Project, utility) -> Non
...
+    def test_writeline_with_different_log_levels(self, caplog):
+        """Test parsing logs with different severity levels."""
+        logger = structlog.stdlib.get_logger("test")
+        handler = _LogOutputHandler(logger, log_parser="singer-sdk")
+
+        error_log = json.dumps(
+            {
+                "level": "error",
+                "pid": 123,
+                "logger_name": "tap-test",
```
[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply tests/meltano/cli/test_invoke.py]: See b887b321d

File: tests/meltano/cli/test_invoke.py [hunk 5]
```diff
@@ -268,3 +271,107 @@ def test_invoke_only_install(self, cli_runner, project: Project, utility) -> Non
...
+    def test_writeline_with_different_log_levels(self, caplog):
+        """Test parsing logs with different severity levels."""
+        logger = structlog.stdlib.get_logger("test")
+        handler = _LogOutputHandler(logger, log_parser="singer-sdk")
+
+        error_log = (
+            '{"level":"error","pid":123,"logger_name":"tap-test","ts":1234567890.0,'
+            '"thread_name":"MainThread","app_name":"singer-sdk","stream_name":null,'
+            '"message":"Error occurred","extra":{}}'
+        )
```
[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply tests/meltano/cli/test_invoke.py]: See b887b321d

**Review by edgarrmondragon — COMMENTED** [review]
PTAL
